### PR TITLE
HT-2244 - support for EMMA METS

### PR DESCRIPTION
**Pull request by respinos**
Add basic support for objects with the EMMA METS type. Scopes enough functionality so EMMA METS can be found and parsed using the existing machinery.

### CODE REVIEW

File: MdpItem/EMMA.pm [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+sub SetSources {
+    my $self = shift;
+    my $root = $self->_GetMetsRoot();
+
+    # NOP
+}
+
+sub GetRemediatedFileId {
```
Comment (aelkiss) [inline MdpItem/EMMA.pm:167]: Doesn't appear to be used by https://github.com/hathitrust/imgsrv/pull/7

File: MdpItem/EMMA.pm [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    }
+    return undef;
+}
+
+sub GetRemediatedFileIds {
+    my $self = shift;
+    my @retval = ();
+    my $fileGrpHash = $self->GetFileGroupMap();
+    foreach my $fileid ( keys %$fileGrpHash ) {
+        if ( $$fileGrpHash{$fileid}{filegrp} eq 'remediatedfile' ) {
```
Comment (aelkiss) [inline MdpItem/EMMA.pm:183]: Is this looking at `fileGrp@USE`? If so it's set to `remediated`, e.g.

```xml
    <METS:fileGrp USE="remediated" ID="FG3">
      <METS:file ID="EMMA00000001" SEQ="00000001" MIMETYPE="application/epub+zip" CREATED="2020-10-25T03:55:46Z" SIZE="1569354" CHECKSUM="edf625d29070896753b8002c996e5041" CHECKSUMTYPE="MD5">
        <METS:FLocat OTHERLOCTYPE="SYSTEM" LOCTYPE="OTHER" xlink:href="u5f94f62cq42.epub"/>
      </METS:file>
    </METS:fileGrp>
```

**Review by aelkiss — COMMENTED** [review]
See comments

File: MdpItem/EMMA.pm (hunk 2 above)
Comment (respinos) [reply MdpItem/EMMA.pm:183]: MdpItem has always tacked _file_ onto the `@USE` value (`imagefile`, `ocrfile`, etc).

File: MdpItem/EMMA.pm (hunk 1 above)
Comment (respinos) [reply MdpItem/EMMA.pm:167]: No, because imgsrv has to currently check for multiple remediated files.

File: Access/Rights.pm [hunk 3]
```diff
@@ -575,7 +575,40 @@ sub get_single_page_PDF_access_status {
...
+        }
+    }
+
+    # clear the error message if $status eq 'allow'
+    $message = '' if ( $status eq 'allow' );
+
+    return ($message, $status);
+}
+
+sub get_remediated_items_access_status {
```
Comment (respinos) [inline Access/Rights.pm:597]: @aelkiss updated access check.

**Review by aelkiss — APPROVED** [review]
Looks good for now. Looking forward to eventually refactoring these to easier-to-comprehend policies...

**Pull request merged**